backend/services/mapping_service.py

```diff
@@ -195,7 +200,11 @@ def add_task_comment(task_comment: TaskCommentDTO) -> TaskDTO:
         """Adds the comment to the task history"""
         task = Task.get(task_comment.task_id, task_comment.project_id)
```

Comment:
Would probably be good to check for the project first. The Not Found error only reflects if the task is not found even if the issue might be a non-existent project.